projects/components/src/multi-select/multi-select.component.ts

```diff
@@ -0,0 +1,164 @@
+            </div>
+          </div>
+        </htc-popover-content>
+      </htc-popover>
+    </div>
+  `
+})
+export class MultiSelectComponent<V> implements AfterContentInit, OnChanges {
+  @Input()
+  public size: SelectSize = SelectSize.Medium;
+
+  @Input()
+  public selected?: V[];
+
+  @Input()
+  public icon?: string;
+
+  @Input()
+  public placeholder?: string;
+
+  @Input()
+  public disabled: boolean = false;
+
+  @Input()
+  public showBorder: boolean = false;
+
+  @Input()
+  public justify?: SelectJustify;
+
+  @Output()
+  public readonly selectedChange: EventEmitter<V[]> = new EventEmitter<V[]>();
+
+  @ContentChildren(SelectOptionComponent)
+  public items?: QueryList<SelectOptionComponent<V>>;
+
+  public selected$?: Observable<SelectOption<V>[] | undefined>;
+  public triggerLabel?: string;
+
+  public get justifyClass(): string {
+    if (this.justify !== undefined) {
+      return this.justify;
+    }
+
+    return this.showBorder ? SelectJustify.Left : SelectJustify.Right;
+  }
+
+  public constructor(private readonly loggerService: LoggerService) {}
+
+  public ngAfterContentInit(): void {
+    this.selected$ = this.buildObservableOfSelected();
+    this.setTriggerLabel();
+  }
+
+  public ngOnChanges(changes: TypedSimpleChanges<this>): void {
+    if (this.items !== undefined && changes.selected !== undefined) {
+      this.selected$ = this.buildObservableOfSelected();
+    }
+    this.setTriggerLabel();
+  }
+
+  private setTriggerLabel(): void {
+    const selectedItems: SelectOptionComponent<V>[] | undefined = this.items?.filter(
+      item => this.selected !== undefined && this.selected?.includes(item.value)
+    );
+    if (selectedItems === undefined || selectedItems.length === 0) {
+      this.triggerLabel = this.placeholder;
+    } else if (selectedItems.length === 1) {
+      this.triggerLabel = selectedItems[0].label;
+    } else {
+      this.triggerLabel = `${selectedItems[0].label} and ${selectedItems.length - 1} more`;
+    }
+  }
+
+  public isSelectedItem(item: SelectOptionComponent<V>): boolean {
+    return this.selected !== undefined && this.selected.filter(value => value === item.value).length > 0;
+  }
+
+  private buildObservableOfSelected(): Observable<SelectOption<V>[] | undefined> {
+    if (!this.items) {
+      return EMPTY;
+    }
+
+    return queryListAndChanges$(this.items).pipe(
+      switchMap(items => merge(of(undefined), ...items.map(option => option.optionChange$))),
+      map(() => this.findItem(this.selected))
+    );
+  }
+
+  public onSelectionChange(item: SelectOptionComponent<V>): void {
+    // If selected in undefined
+    if (this.selected === undefined) {
+      this.selected = [item.value];
+    } else {
+      const selectedItems: V[] = [...this.selected];
+      if (selectedItems.includes(item.value)) {
+        // Remove if already selected
+        this.selected = selectedItems.filter(value => value !== item.value);
```

Comment:
feels like we have 3 or 4 places in this component where we are doing a filter/include type check to map between selections and items - we should be able to define that in one spot.